rtl_tests.c

```diff
@@ -98,35 +99,36 @@ void test_RtlAppendUnicodeStringToString(){
     BOOL tests_passed = 1;
     print_test_header(func_num, func_name);
 
-    // wcslen does not work because nxdk fails to find wchar.h
-    const WCHAR src_text[] = L"Xbox";
-    const uint8_t num_chars_in_src = 4;
-    WCHAR buffer[num_chars_in_src * 2];
-    const uint8_t num_buf_bytes = sizeof(buffer);
     UNICODE_STRING src_str, dest_str;
+    RtlInitUnicodeString(&src_str, L"Xbox");
+    const size_t num_chars_in_src = wcslen(src_str.Buffer);
+    WCHAR buffer[(num_chars_in_src + 1) * 2]; // Add 1 to hold the \0 end character
```

Comment:
The WCHAR type is already 2 bytes long, there should be no need to multiply by 2

Reply:
That code creates a buffer that is 2x the number of wchars (with the NULL terminator \0 taken into account with the +1) as the src. `num_chars_in_src = 4` in this case, so the code creates a buffer that is 10 WCHAR characters large. Big enough to hold src_str.Buffer twice as src_str.Buffer is 5 WCHAR characters large.

Reply:
There should probably be a comment which explains that this will have 2 concatenated copies of the `src_str`. This is indeed very confusing, especially with the zero-termination comment being less relevant [as it's more standard] than the multiplication [which is unique to the use-case].